### ISSUE TITLE
Ansi-Tests: Condition of type: SIMPLE-PROGRAM-ERROR: Not enough arguments - you provided 3 and 4 are required

### ISSUE THREAD

**Issue by kpoeck** [drmeister/clasp#422]
````
karsten-poecks-macbook-pro:clasp2 karstenpoeck$ build/boehm/cclasp-boehm
../../src/main/main.cc:298 Entered try block typeid(ExitProgramException) -> 4403693936
../../src/main/main.cc:299   again try block typeid(ExitProgramException) -> 4403693936
Starting Boehm cclasp 0.4.0-3186-g345b16d6d ... loading image... it takes a few seconds
Top level in: #<PROCESS TOP-LEVEL @0x10aaa7e09>.
COMMON-LISP-USER> (multiple-value-list (typep (make-array 2) '(SIMPLE-ARRAY * 2)))

Condition of type: SIMPLE-PROGRAM-ERROR
Not enough arguments - you provided 3 and 4 are required

Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[1] ERROR. In: #<PROCESS TOP-LEVEL @0x10aaa7e09>.
 File: #P"/Users/karstenpoeck/lisp/compiler/clasp2/src/core/lisp.cc" (Position #90542)
COMMON-LISP-USER>> :b
0 UNIVERSAL-ERROR-HANDLER
1 (ERROR CORE:SIMPLE-PROGRAM-ERROR :FORMAT-CONTROL "Not enough arguments - you provided 3 and 4 are required" :FORMAT-ARGUMENTS NIL)
2 APPLY
3 SIGNAL-SIMPLE-ERROR
4 MAYBE-GEN-PRIMITIVE-TYPE-CHECK
5 GEN-ARRAY-TYPE-CHECK
6 LAMBDA
7 LAMBDA
8 REPLACE-TYPEQ
9 LAMBDA
10 LAMBDA
11 {INVOKE-WITH-METER ((METER T))}
12 LAMBDA
13 {INVOKE-WITH-METER ((:AROUND) (BASIC-METER T))}
14 LAMBDA
15 {INVOKE-WITH-METER ((:AROUND) (SIZE-METER T))}
16 REDUCE-TYPEQS
17 MY-HIR-TRANSFORMATIONS
18 COMPILE-FORM-TO-MIR
19 LAMBDA
20 LAMBDA
21 LAMBDA
22 LAMBDA
23 LAMBDA
24 LAMBDA
25 LAMBDA
26 LAMBDA
27 LAMBDA
28 LAMBDA
29 LAMBDA
30 CCLASP-COMPILE-TO-MODULE-WITH-RUN-TIME-TABLE
31 LAMBDA
32 LAMBDA
33 CCLASP-COMPILE*
34 COMPILE-WITH-HOOK
35 LAMBDA
36 LAMBDA
37 LAMBDA
38 LAMBDA
39 LAMBDA
40 LAMBDA
41 LAMBDA
42 LAMBDA
43 LAMBDA
44 DO-COMPILATION-UNIT
45 LAMBDA
46 LAMBDA
47 LAMBDA
48 LAMBDA
49 LAMBDA
50 DO-COMPILATION-UNIT
51 LAMBDA
52 LAMBDA
53 LAMBDA
54 LAMBDA
55 LAMBDA
56 LAMBDA
57 LAMBDA
58 COMPILE-IN-ENV
59 LAMBDA
60 LAMBDA
61 CCLASP-COMPILE-IN-ENV
62 LAMBDA
63 {CCLASP-EVAL-WITH-ENV ((T T))}
64 APPLY
65 PERFORM-OUTCOME
66 DO-DISPATCH-MISS.MULTIPLE-VALUE-BIND.LAMBDA
67 INVALIDATED-DISPATCH-FUNCTION
68 LAMBDA
69 LAMBDA
70 LAMBDA
71 CATCH-LAMBDA
72 LAMBDA
73 LAMBDA
74 LAMBDA
75 LAMBDA
76 LAMBDA
77 LAMBDA
78 LAMBDA
79 LAMBDA
80 TPL
81 LAMBDA
82 LAMBDA
83 LAMBDA
84 LAMBDA
85 LAMBDA
86 LAMBDA
87 LAMBDA
88 LAMBDA
89 LAMBDA
90 LAMBDA
91 LAMBDA
92 CATCH-LAMBDA
93 LAMBDA
````

**Comment by kpoeck** [comment]
The error is in (defun gen-array-type-check (object element-type dimensions simple-only-p pro con) ..)
where in 1 call to maybe-gen-primitive-type-check 1 parameter is missing.

I think the fix is the following (marked with kpoeck). Will make a PR
````
(defun gen-array-type-check (object element-type dimensions simple-only-p pro con)
  (let* ((dimensions (if (integerp dimensions) (make-list dimensions :initial-element '*) dimensions))
         (rank (if (eq dimensions '*) '* (length dimensions)))
         (simple-vector-type
           (if (eq element-type '*)
               'core:abstract-simple-vector
               (simple-vector-type element-type)))
         (complex-vector-type
           (case element-type
             ((base-char) 'core:str8ns)
             ((character) 'core:str-wns)
             ((bit) 'core:bit-vector-ns)
             (t nil)))
         (simple-mdarray-type
           (if (eq element-type '*)
               'core:simple-mdarray
               (simple-mdarray-type element-type)))
         (mdarray-type
           (if (eq element-type '*)
               'core:mdarray
               (complex-mdarray-type element-type))))
    (let ((pro (if (eq dimensions '*)
                   pro
                   (loop for dim in dimensions
                         for i from 0
                         for test = (gen-dimension-check object i dim pro con)
                           then (gen-dimension-check object i dim test con)
                         finally (return test)))))
      (cond (simple-only-p
             (cond ((eql rank 1)
                    (maybe-gen-primitive-type-check
                     object simple-vector-type pro con))
                   ((eq rank '*)
                    (maybe-gen-primitive-type-check
                     object simple-vector-type pro
                     (maybe-gen-primitive-type-check
                      object simple-mdarray-type pro con)))
                   (t
                    (maybe-gen-primitive-type-check
                     object simple-mdarray-type
                     (gen-rank-check object rank pro con)
                     ;;;kpoeck
                     con))))
            (t
             (cond ((eql rank 1)
                    (maybe-gen-primitive-type-check
                     object simple-vector-type pro
                     (if complex-vector-type ; probably next most likely.
                         (maybe-gen-primitive-type-check
                          object complex-vector-type pro
                          (maybe-gen-primitive-type-check
                           object mdarray-type
                           (gen-rank-check object rank pro con)
                           con))
                         ;; no complex vector type, so just mdarray
                         (maybe-gen-primitive-type-check
                          object mdarray-type
                          (gen-rank-check object rank pro con) con))))
                   ((eq rank '*)
                    ;; just the above without the rank check.
                    (maybe-gen-primitive-type-check
                     object simple-vector-type pro
                     (if complex-vector-type
                         (maybe-gen-primitive-type-check
                          object complex-vector-type pro
                          (maybe-gen-primitive-type-check
                           object mdarray-type pro con))
                         (maybe-gen-primitive-type-check
                          object mdarray-type pro con))))
                   (t ;; checking for some multidimensional type
                    (maybe-gen-primitive-type-check
                     object mdarray-type
                     (gen-rank-check object rank pro con) con))))))))
````

**Comment by kpoeck** [comment]
Should be fixed by PR #423

**Comment by kpoeck** [comment]
Fixed in dev with the merged PR
